ambry-store/src/test/java/com.github.ambry.store/HardDeleterTest.java

```diff
@@ -305,3 +306,76 @@ public void testHardDelete() {
     }
   }
 }
+
+class MockIndex extends PersistentIndex {
```

Comment:
A lot of the functions/constructors are no longer in use. Could you remove them? Also remove `public` from the keywords (can even make this a private class since it is needed only by this test class)